synapse/rest/client/v2_alpha/sync.py

```diff
@@ -356,6 +356,12 @@ def _rollback_state_for_timeline(state, timeline):
 
```

Comment:
It's a shame the log at lines 354/355 will be a lie if the prev_* keys are missing :(